Userland/Libraries/LibGfx/Bitmap.h

```diff
@@ -171,6 +171,7 @@ class Bitmap : public RefCounted<Bitmap> {
     BitmapFormat m_format { BitmapFormat::Invalid };
     bool m_data_is_malloced { false };
     Core::AnonymousBuffer m_buffer;
+    Function<void()> m_destruction_callback;
```

Comment:
Feels like there's an opportunity to move the 'data is malloc'd' logic into this destruction callback idea.